libs/sdk-core/src/breez_services.rs

```diff
@@ -1553,15 +1553,18 @@ impl Receiver for PaymentReceiver {
             });
         }
 
-        // create the large amount invoice
-        let raw_invoice_with_hint =
-            add_lsp_routing_hints(invoice.bolt11.clone(), lsp_hint, amount_sats * 1000)?;
+        // We only create a new invoice if we need to add the lsp hint or change the amount
+        if lsp_hint.is_some() || amount_sats != destination_invoice_amount_sats {
```

Comment:
It seems to me the first created invoice never has routing hints, so won't this always happen as the LSP routing hint **has** to be added?

Reply:
For the next invoices (when there is enough liquidity) we won't enter this block as gl will populate the hints.
Only for invoices that opens channels will be encoded.
On some rate cases gl doesn't populate the hints even though the channel exists and this is why we add them manually